catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/NsiliEndpoint.java

```diff
@@ -106,25 +107,45 @@ public int getMaxNumResults() {
         return maxNumResults;
     }
 
+    private void setLibraryQuerySources(Set<String> querySources) {
+        Optional.ofNullable(library)
+                .ifPresent(l -> l.setQuerySources(querySources));
+    }
+
     public void setMaxNumResults(int maxNumResults) {
         this.maxNumResults = maxNumResults;
-        if (library != null) {
-            library.setMaxNumResults(maxNumResults);
-        }
+        Optional.ofNullable(library)
+                .ifPresent(l -> l.setMaxNumResults(maxNumResults));
     }
 
-    public List<String> getQuerySources() {
+    @Override
+    public Set<String> getQuerySources() {
         return querySources;
     }
 
-    public void setQuerySources(List<String> querySources) {
-        this.querySources.clear();
+    public void setQuerySources(Set<String> querySources) {
+        //TODO? check if framework.getSourceIds().containsAll(querySources)
         if (querySources != null) {
+            this.querySources.clear();
             this.querySources.addAll(querySources);
+            setLibraryQuerySources(querySources);
+        }
+    }
+
+    @Override
+    public void addQuerySource(String sourceId) {
+        //TODO? check if framework.getSourceIds().contains(sourceId)
```

Comment:
@troymohl Should framework.getSourceIds().contains(sourceId) be checked before a sourceId is added to querySources?